base/ltcmdhooks.dtx

```diff
@@ -274,8 +274,8 @@
 %
 % Command hooks are intended to be added to document commands, which are
 % typically defined globally. As such, adding a hook is a global operation,
-% even if the command was previously only locally-defined. The
-% \emph{content} of hooks respect \TeX{} groupings
+% even if the command was previously only locally-defined. Addition of
+% material to hooks is also global.
 %
```

Comment:
Just curious whether you want to mention, that it would still be possible to achieve different local behavior within a group by one level of indirection
```latex
\NewDocumentCommand\foo{foo executed}{}
\newcommand\afterfoo{with global hook}
\AddToHook{cmd/foo/after}{\afterfoo}

\foo

\begingroup
\renewcommand\afterfoo{with grouping hook}
\foo
\endgroup
```